### DISCUSSION TITLE
An "invalid response was received from the upstream server" error when querying auth

### DISCUSSION THREAD

**Discussion by kangmingtay** [supabase/supabase#20722] [Troubleshooting]
_This is a copy of a troubleshooting article on Supabase's docs site. It may be missing some details from the original. View the [original article](https://supabase.com/docs/guides/troubleshooting/an-invalid-response-was-received-from-the-upstream-server-error-when-querying-auth-RI4Vl-)._

If you are observing an "invalid response was received from the upstream server" error when making requests to Supabase Auth, it could mean that the respective service is down. One way to confirm this is to visit the [logs explorer](https://supabase.com/dashboard/project/_/logs/explorer) and look at the auth logs to see if there are any errors with the following lines:

*   `running db migrations: error executing migrations/20221208132122_backfill_email_last_sign_in_at.up.sql`

We're currently investigating an issue where the tables responsible for keeping track of migrations ran by Auth (`auth.schema_migrations`) are not being restored properly, which leads to the service(s) retrying those migrations. In such cases, migrations which are not idempotent will run into issues.

We've documented some of the migrations that run into this issue and their corresponding fix here:

## Auth: `operator does not exist: uuid = text`

Temporary fix: Run `insert into auth.schema_migrations values ('20221208132122');` via the [SQL editor](https://supabase.com/dashboard/project/_/sql/new) to fix the issue.

If the migration error you're seeing looks different, reach out to [supabase.help](https://supabase.help/) for assistance.

**Comment by FutureFutureTo** [comment]
thanks for this - any progress on it being fixed? im in the early phases of an app and need to reset the auth tables often enough in production

**Reply by kangmingtay** [reply]
you shouldn't be resetting the `auth.schema_migrations` table as that table is used by the auth service to detect which migrations have be run already

**Comment by Demizon3433** [comment]
I just ran into this issue on two of my supabase projects. The temporary fix works. Thanks for that!

I'd like to paste in the exact error from the database log to make this issue easier to find by just pasting it into the browser.

{"level":"fatal","msg":"running db migrations: error executing migrations/20221208132122_backfill_email_last_sign_in_at.up.sql, sql: -- previous backfill migration left last_sign_in_at to be null, which broke some projects\n\ndo $$\nbegin\nupdate auth.identities\n  set last_sign_in_at = '2022-11-25'\n  where\n    last_sign_in_at is null and\n    created_at = '2022-11-25' and\n    updated_at = '2022-11-25' and\n    provider = 'email' and\n    id = user_id::text;\nend $$;\n: ERROR: operator does not exist: uuid = text (SQLSTATE 42883)","time":"2024-04-30T12:41:58Z"}

**Comment by marcnajera** [comment]
Had the same problem this week. The fix worked (thanks!) but it was quite hard to figure out what was going on. Do you have updates on this?

**Reply by kangmingtay** [reply]
you can check your project's auth logs which should reflect the error (https://supabase.com/dashboard/project/_/logs/auth-logs) 

it seems like the auth.schema_migrations table is being cleared, which causes the auth service to attempt to rerun all the migrations. This particular migration fails because it's not idempotent, since it's a data migration.

We're still investigating the issue but it seems like users experiencing this might be using a tool (prsima or supabase CLI) which resets the tables in the auth schema, perhaps when syncing local dev to remote, and this causes the rows in the auth.schema_migrations table to be cleared.

**Reply by tananga** [reply]
@kangmingtay Im often executes "supabase db reset --linked" and I have the same issue. https://github.com/supabase/supabase/issues/26247

**Reply by kangmingtay** [reply]
@Tananga before you ran `supabase db reset --linked`, did you check if the auth service was healthy or if it was down already?

**Reply by tananga** [reply]
I didnt check it. But there were 2 commands which Im executing very often 

Supabase db reset --linked

And from time to time:

`delete from supabase_migrations.schema_migrations;`

**Reply by sweatybridge** [reply]
`npx supabase@beta db reset --linked` should fix this issue. More details here https://github.com/orgs/supabase/discussions/20722#discussioncomment-9601951

**Reply by ThatGuySam** [reply]
> `npx supabase@beta db reset --linked` should fix this issue. More details here https://github.com/orgs/supabase/discussions/20722#discussioncomment-9601951

Tried this today with that command which resolved to 1.172.4: No luck. 

Ended up having to migrate over to a brand new database instance for that version. 

Would still love a solution in case this happens on an instance I can't migrate the data for.

**Comment by tobiastornros** [comment]
Adding screenshot. Our staging project got this error after random shutdown today. Any progress on a fix? 

![Screenshot 2024-04-30 at 12 30 07](https://github.com/supabase/supabase/assets/17402497/f20d3c15-a535-43b0-a012-885dc66056d5)

Adding errors for search visibility: 
`error executing migrations/20221125140132_backfill_email_identity.up.sql ...`
`error executing migrations/20221208132122_backfill_email_last_sign_in_at.up.sql ...`

Fix:
`insert into auth.schema_migrations values ('20221125140132');`
`insert into auth.schema_migrations values ('20221208132122');`

**Reply by kangmingtay** [reply]
It seems like the auth.schema_migrations table is being cleared, which causes the auth service to attempt to rerun all the migrations. This particular migration fails because it's not idempotent, since it's a data migration.

We're still investigating the issue but it seems like users experiencing this might be using a tool (prsima or supabase CLI) which resets the tables in the auth schema, perhaps when syncing local dev to remote, and this causes the rows in the auth.schema_migrations table to be cleared.

**Reply by tobiastornros** [reply]
Correct. Via use supabase cli to reset staging project every night and after that restores public schema with data from snaplet.

**Reply by tobiastornros** [reply]
Just fyi. It happens when the auth service restart and not when we use supabase cli to reset linked project.

**Reply by kangmingtay** [reply]
@tobiastornros does this happen everytime you restart the auth service? (i.e when you restart your project through the dashboard and then use supabase cli to reset the linked project?) 

or does this happen randomly?

**Reply by tobiastornros** [reply]
Randomly. You can see in the screenshot above that the auth service get shutdown signal at 11:31. We run the reset command around 03:00 every night.

**Reply by kangmingtay** [reply]
@tobiastornros can you please send us a ticket for that particular project at https://supabase.help and link this thread in the ticket so we can dig into it?

**Reply by tobiastornros** [reply]
@kangmingtay Done

**Comment by tmountain** [comment]
Same bug over here. Can someone explain what is happening here?

<img width="775" alt="Screenshot 2024-05-01 at 3 47 27 PM" src="https://github.com/supabase/supabase/assets/135297/498a14a6-c01d-40ac-b35d-c334c2b26563">

**Reply by kangmingtay** [reply]
It seems like the `auth.schema_migrations` table is being cleared, which causes the auth service to attempt to rerun all the migrations. This particular migration fails because it's not idempotent, since it's a data migration. 

We're still investigating the issue but it seems like users experiencing this might be using a tool (prsima or supabase CLI) which resets the tables in the auth schema, perhaps when syncing local dev to remote, and this causes the rows in the `auth.schema_migrations` table to be cleared.

**Reply by tmountain** [reply]
We have a Github action doing this when we deploy to our staging DB.

```
jobs:
  build:
    runs-on: ubuntu-latest
    env:
      supabase_db_url: ${{ secrets.STAGING_DB_URL }}

    steps:
      - uses: actions/checkout@v4

      - uses: supabase/setup-cli@v1
        with:
          version: latest

      - run: yes | supabase db reset --db-url "$supabase_db_url"
```

[kiwicopple](https://www.reddit.com/user/kiwicopple) from Supabase indicated that this was okay on Reddit. It's now an important part of our dev workflow. It'd be really nice to have this supported without having sporadic outages on our instance.

**Reply by FutureFutureTo** [reply]
Yea, my particular use case here is that I am still working on my local schema and would like to be able to have my production match exactly once I have landed on what it is I have decided. I understand that once in production, this is bad practice, but I'm still tweaking and would like the flexibility to easily have production match exactly local.

**Reply by kangmingtay** [reply]
@tmountain does this always happen every time to CI runs or does it happen randomly?

**Reply by tmountain** [reply]
@kangmingtay It's random. Feels like a race condition of some kind.

**Reply by sweatybridge** [reply]
I've updated CLI beta channel to not reset the auth migrations table. You can update it on GHA like this

```
      - uses: supabase/setup-cli@v1
        with:
          version: 1.172.1
```

The next stable release will be on 1 June. After that, you can revert back to `version: latest`.

**Comment by sebmor** [comment]
Had it as well on production db 
![Screenshot 2024-05-01 at 22 50 37](https://github.com/supabase/supabase/assets/405407/9cec5a95-1e36-4710-962f-17a3eed68ce4)

**Comment by etzelc** [comment]
I've seen this on our testing instance as well and applied the temporary fix successfully. Is there any Github issue tracking the initial problem of the untracked Auth migrations?

**Reply by kangmingtay** [reply]
It seems like the auth.schema_migrations table is being cleared, which causes the auth service to attempt to rerun all the migrations. This particular migration fails because it's not idempotent, since it's a data migration.

We're still investigating the issue but it seems like users experiencing this might be using a tool (prsima or supabase CLI) which resets the tables in the auth schema, perhaps when syncing local dev to remote, and this causes the rows in the auth.schema_migrations table to be cleared.

**Reply by etzelc** [reply]
Yes, we are using supabase cli to reset our testing instance. If needed I can provide more details on the exact command order on Thursday.

**Reply by kangmingtay** [reply]
that would be great @etzelc

**Reply by etzelc** [reply]
We've noticed that the issue occasionally arises on our testing instance. The testing instance is the only instance which is reseted during the deployment, because it is used to try new features, migrations, etc. from various feature branches. Here's an outline of the deployment steps:
```
- npx supabase link --project-ref ABC123XYZ789
- npx supabase db reset --linked # there are migrations in supabase/migrations and test data in supabase/seed.sql
- npx supabase secrets set ...
- npx supabase functions deploy # and some more commands to delete uncommitted (old) upstream functions
```
The supabase CLI version is either the newest stable release or one stable version released within the past weeks.

**Reply by sweatybridge** [reply]
We have updated CLI on 1 June so it should no longer reset the auth migrations table. See https://github.com/orgs/supabase/discussions/20722#discussioncomment-9601951

**Reply by etzelc** [reply]
Thanks for your response and the fix. We upgraded our CLI yesterday and haven't experienced the issue since.

**Comment by robertbodley** [comment]
I am experiencing the same issue (the temp fix worked)

**Comment by indie-rok** [comment]
Thanks! Can you point me on how to properly reset production db? (if supabase db reset --linked is not a good idea)

**Reply by sweatybridge** [reply]
The reset command was bugged. It is now fixed in CLI v1.172.2 so it no longer clears the auth migrations table. See https://github.com/orgs/supabase/discussions/20722#discussioncomment-9601951

**Reply by fvermaut** [reply]
@sweatybridge thanks for the update - but was it really the root cause?   Because for me the issue happened randomly, not after a db reset

**Comment by rookledookle** [comment]
Experienced this issue today, fixed worked, but I'm wondering:

What did I do / what happened that caused this? I don't believe I made any changes since it was last working (yesterday) vs getting the error today. 

Not sure if these details are relevant, posting in case it's helpful: 
- I was testing my login flows on my staging and prod environment (both separate supabase projects) and doing them in the same browser without clearing cache when this happened
- I set up my prod db just yesterday by pulling from my staging db to local, and then doing a supabase link, then a supabase db reset --linked via the CLI

**Reply by rookledookle** [reply]
My main worry is if this is going to happen again / how do I prevent it as breaking auth on prod is really scary

**Reply by rookledookle** [reply]
Also, slight aside: am I able to get these types of errors proactively reported via alerting somehow? It's not captured in Sentry but I don't know if that's due to my instrumentation or if it just won't catch such errors.

**Reply by kangmingtay** [reply]
we don't offer alerting directly from the platform but there is a health check endpoint `https://your_project_ref.supabase.co/auth/v1/health` that you can periodically ping to get alerts if the service is down

> I set up my prod db just yesterday by pulling from my staging db to local, and then doing a supabase link, then a supabase db reset --linked via the CLI

when you run supabase link + supabase db reset, are those ran against the staging db or production db?

**Reply by rookledookle** [reply]
I do a supabase db pull from staging, and then a supabase db link (to production) and supabase db reset --linked (production) 

Thanks! Let me know if you need more info

**Reply by sweatybridge** [reply]
We have updated CLI in release v1.172.2 so it no longer resets auth migrations table.

**Comment by LeMark0** [comment]
In case the quick fix is not working, one possible reason is that GOTRUE tries to add the record to the `public.schema_migrations` table instead of `auth.schema_migrations`.
To fix this, you'll need to either add the record `20221208132122` to `public.schema_migrations` or change the search order for your DB:

```
ALTER ROLE supabase_auth_admin SET search_path TO auth,public;
```

**Comment by Zeulni** [comment]
Had the same issue in production today and thus a downtime of 1-2 hours. The quick fix solved it. 

Please prioritize this issue!

**Comment by fvermaut** [comment]
I also had this issue happen randomly today as well on staging.  

Here's a screenshot of the logs (sorry didn't find a way to export it - the CSV export seems to export something different):
<img width="692" alt="image" src="https://github.com/supabase/supabase/assets/43007270/9464b2a8-4b62-4545-bd6c-c04964199e25">

I agree ith Zeulni: please prioritize this, considering it can happen so randomly, and basically completely breaks auth until the fix is applied.

**Reply by etzelc** [reply]
Have you seen and tested this possible solution? It was just posted today.
[https://github.com/supabase/supabase/discussions/20722#discussioncomment-9601951](https://github.com/supabase/supabase/discussions/20722#discussioncomment-9601951)

**Reply by fvermaut** [reply]
For me the issue was not triggered by a CI, or any push action I did from the CLI.   I really don't know why there has been a shut down at 13:45:44 (which appears to be the trigger).

**Reply by jchaselubitz** [reply]
I had this issue at exactly the same time (13:49:07)

**Comment by primalyeti** [comment]
Had this issue pop on me **again**, seemingly randomly when i updated my functions.

The fix was running `insert into auth.schema_migrations values ('20221208132122');`, same query i had to run a few days ago to fix this bug.

This should be a high priority as it completely destroys authentication

**Reply by stefan-girlich** [reply]
> The fix was running `insert into auth.schema_migrations values ('20221208132122');`, same query i had to run a few days ago to fix this bug.

Thanks @primalyeti , this worked for me whereas `supabase db reset --linked` did not. Just experienced this issue for the first time just now when trying to create a user via Supabase Studio (cloud). Fully agree that this needs to be prioritized.

Concerning `supabase db reset --linked`: Wouldn't this erase the remote/production database? I'm not sure how this would then work as a general fix.

**Reply by sweatybridge** [reply]
The reset command no longer clears auth migrations table. But it does not restore the auth table if it's already cleared. Please don't run it as a general fix as it erases data from your remote project.

**Comment by hf** [comment]
Are folks using Prisma or some other ORM? Please react 👍 if yes.

**Comment by hf** [comment]
We're gonna drop those two migrations probably, but still no idea why that table is getting cleared. https://github.com/supabase/auth/pull/1603

**Reply by GaryAustin1** [reply]
https://github.com/orgs/supabase/discussions/20722#discussioncomment-9601951
Maybe?

**Comment by luong-komorebi** [comment]
We are also impacted by this issue yesterday. I dont think we have made any changes to supabase recently nor do we run any migration in the last week 

Project id is `qmikvpynjsthwajkswhs` if someone at supabase want to check our logs and event trails

**Comment by MadMango** [comment]
Getting this on project id `oqhsplvvzhlidhxvduaf`, seemingly random.
I tried resetting the linked db but no luck.

It seems that migrations are a bit funky recently, I was getting some changes that I didn't make included in schema diffs recently as well.

**Comment by al3rez** [comment]
We had a demo to a client ytd, I ended up demoing using my local machine, but the temp fix worked for now... how can we prevent this sort of issues? Is this production ready?

**Comment by VitaliyHula** [comment]
I have the same issue
This happened after the first user registered via email and password.
Until that moment, we had no users who were registered via email and password

**Reply by nechita** [reply]
> I have the same issue This happened after the first user registered via email and password. Until that moment, we had no users who were registered via email and password

Something very similar happened to me as well, the difference is that the migration which is failing is a different one.

`running db migrations: error executing migrations/20230131181311_backfill_invite_identities.up.sql`

The situation happened exactly after creating a user with email and password.

This was extremely frustrating, because I didn't change anything in the codebase, but somehow nothing was working :(.

**Comment by shahrierNafis** [comment]
the temporary fix worked with auth but realtime is still down 
![image](https://github.com/supabase/supabase/assets/102902752/22960466-2b2d-47f6-9c5c-5f5ff5d4da9d)

**Comment by warrenbhw** [comment]
I'm still hitting this issue. Any updates on the cause and a permanent fix?

**Comment by nechita** [comment]
I've applied the temporary fix suggested in the thread but is updating the CLI a solid solution to prevent this from happening again when doing `supabase db push` to linked projects?

**Comment by hojeffrey** [comment]
Doing local dev and ran into this issue, cant even run my containers. I stopped the containers and could not start it back up with the same errors. Am I doing something wrong here? Had to clear out my local volumes but that destroyed all the progress I made without even being able to make a migration file

**Reply by warrenbhw** [reply]
For now, the best answer is just to use the temporary fix in the original post. Quoting here:

> Temporary fix: Run `insert into auth.schema_migrations values ('20221208132122')`; via the [sql editor](https://supabase.com/dashboard/project/_/sql/new) to fix the issue.

**Reply by warrenbhw** [reply]
It sounds like there is some failing migration in the supabase-setup migrations, perhaps because there's some edge case where supabase is trying to create or delete the same resource twice. Thus, subsequent migrations fail and the auth schema doesn't get set up as it should.

I am only hitting this error on rare occasions when I do a reset of my nonlocal supabase db. In the cases where this happens, I just have to run this sql statement one time to fix.

Hopefully the supabase team can find a permanent fix, but I did find that this unblocked me and did not lead to any subsequent issues.

**Comment by anarkrypto** [comment]
Same issue here running locally.

I tried `insert into auth.schema_migrations values ('20221208132122')`

But it returns: `Error: ERROR: 23505: duplicate key value violates unique constraint "schema_migrations_pkey" DETAIL: Key (version)=(20221208132122) already exists.`

The strangest thing is that sometimes it starts working on its own, but sometimes it doesn't come back at all.

So I try to stop the container... ok.
Then I try to start again and I get this error:
```
You are running outdated service versions locally:
supabase/gotrue:v2.153.0 => v2.155.3
supabase/storage-api:v1.6.4 => v1.6.8
Run supabase link to update them.
Stopping containers...
failed to create docker container: Error response from daemon: Conflict. The container name "/supabase_rest_my_supabase" is already in use by container "2a4ff47ac5a51e4353fbd44b3e9137c9c1045615992c57b0c95942aaeab2d295". You have to remove (or rename) that container to be able to reuse that name.
```

Then I try to start again and works.

Supabase 1.183.5

@sweatybridge

**Comment by jason1610** [comment]
Any updates? The temporary fix works but the fact that this happens randomly in production is nutty.

**Reply by kangmingtay** [reply]
@jason1610 we're working on locking down the permissions on the auth.schema_migrations table to only allow the supabase_auth_admin role to perform DDL statements on that table - this should help to prevent external tools from deleting / truncating the schema migrations table

**Comment by anarkrypto** [comment]
For me it only resolved after I deleted all docker containers, all docker images and all docker volumes.

**Reply by anarkrypto** [reply]
The error is back 😭

**Reply by kangmingtay** [reply]
@anarkrypto what's your Supabase CLI version? are you using prisma / `@snaplet/seed` ?

**Comment by blytheaw** [comment]
I started facing this issue recently and found this thread. I was already on a newer version of the CLI so I wasn't sure what was causing it. After some investigation, it appears that if using the `@snaplet/seed` package's $resetDatabase() function and you have your client configured to inspect the full `auth` schema, it will erase the data in the `auth.schema_migrations` table, causing the same issue discussed in this thread.

For anyone in this boat, make sure your seed.config.ts doesn't include all of the `auth` schema. Use this config:

```typescript
  select: [
    // We don't alter any extensions tables that might be owned by extensions
    "!*",
    // We want to alter all the tables under public schema
    "public*",
    // We also want to alter some of the tables under the auth schema
    "auth.users",
    "auth.identities",
    "auth.sessions",
  ]
```

Ref: https://snaplet-seed.netlify.app/seed/integrations/supabase#2-set-up-seed

Better yet, if your seeding doesn't require any knowledge of the auth schema, don't include it at all.

Feedback for Supabase: Obviously users should be wary of using external tools that can delete things they shouldn't, but I also think Supabase should not let users so easily break a managed service. If this table is so sensitive it should be locked down and not touchable by end users, or extra care should be taken to make these migrations idempotent, as others have suggested.

**Reply by kangmingtay** [reply]
>I also think Supabase should not let users so easily break a managed service. If this table is so sensitive it should be locked down and not touchable by end users, or extra care should be taken to make these migrations idempotent, as others have suggested.

@blytheaw agreed, we're working on locking down the permissions on the `auth.schema_migrations` table to only allow the `supabase_auth_admin` role to perform DDL statements on that table - this should help to prevent external tools from deleting / truncating the schema migrations table

thanks for identifying the issue with the `@snaplet/seed` package btw!

**Comment by mlawlerau** [comment]
I had this issue on my stage db and I had to make 2 manual edits to the database to make it work:
insert into auth.schema_migrations values ('20221208132122');
insert into auth.schema_migrations values ('20240806073726');

What is causing the migration errors - it seems lots of folks are encountering the issue and there isn't a real answer from supabase apart from "hack your database to resolve"?

How can we make sure this issue doesn't happen again?

**Reply by kangmingtay** [reply]
@mlawlerau we're working on locking down the permissions on the auth.schema_migrations table to only allow the supabase_auth_admin role to perform DDL statements on that table - this should help to prevent external tools from deleting / truncating the schema migrations table

**Comment by paulcalcraft** [comment]
To fix this today I had to run:
```insert into auth.schema_migrations values ('20230131181311');```

(different migration ID)

In general, it seems like you can look for the migration error in your auth logs https://supabase.com/dashboard/project/_/logs/auth-logs and just copy the timestamp/migration ID of the failing migration, and insert that. It fixed it straight away for me.

My error was for `{"level":"fatal","msg":"running db migrations: error executing migrations/20230131181311_backfill_invite_identities.up.sql` hence inserting `20230131181311`

**Reply by kangmingtay** [reply]
hi @paulcalcraft, can you please check if you're using the latest version of supabase CLI?

**Comment by NicoCaldo** [comment]
This happened to me running `supabase start` on a clean installation. Supabase CLI updated on the latest version

**Reply by kangmingtay** [reply]
@NicoCaldo can you specify which version is this? also, are you using prisma or the @snaplet/seed package?

**Comment by flurret** [comment]
This just happened to me on a supabase hosted server, with id: gthsgmwqhrnaagwurdbv. 
I think it might have been caused by me changing STMP settings which is odd. 
I had both 20221208132122_backfill_email_last_sign_in_at.up.sql and 20240806073726_drop_uniqueness_constraint_on_phone.up.sql causing issues until I did the insert fix.

It seems like this is still a major problem, agree with all the above comments saying this should be fixed with high priority. 
I'm just glad this db was not my production db, however it does make me nervous to host on supabase in the future.

**Comment by NicoCaldo** [comment]
At least it should display what type of error occurred and where, not a generic "an invalid response from upstream"

**Reply by kangmingtay** [reply]
@NicoCaldo the errors can be seen in your project's auth logs - https://supabase.com/dashboard/project/_/logs/auth-logs

**Comment by thor-vemove** [comment]
Hello all and hello @kangmingtay,
First of all, I want to thank the Supabase team for the fantastic product and ongoing support.

Since a support engineer forwarded this discussion to us, I wanted to share our experience with a similar issue.

We also experienced the issue with `running db migrations: error executing migrations/20221125140132_backfill_email_identity.up.sql`, but in our case it is caused by `null value in column "provider_id" of relation "identities"`. Here’s the full error message:
```json
{"level":"fatal","msg":"running db migrations: error executing migrations/20221125140132_backfill_email_identity.up.sql, sql: -- backfill the auth.identities column by adding an email identity \n-- for all auth.users with an email and password \n\ndo $$\nbegin\n\tinsert into auth.identities (id, user_id, identity_data, provider, last_sign_in_at, created_at, updated_at)\n\tselect id, id as user_id, jsonb_build_object('sub', id, 'email', email) as identity_data, 'email' as provider, null as last_sign_in_at, '2022-11-25' as created_at, '2022-11-25' as updated_at\n\tfrom auth.users as users\n\twhere encrypted_password != '' and email is not null and not exists(select user_id from auth.identities where user_id = users.id);\nend;\n$$;\n: ERROR: null value in column \"provider_id\" of relation \"identities\" violates not-null constraint (SQLSTATE 23502)","time":"2024-10-23T07:06:09Z"}
```

We temporarily resolved the issue by adding the serial of the migration to the auth schema history. However, we encountered another issue: `error executing migrations/20240806073726_drop_uniqueness_constraint_on_phone.up.sql` because `relation "unique_phone_factor_per_user" already exists`. Here’s the full error message:
```json
{"level":"fatal","msg":"running db migrations: error executing migrations/20240806073726_drop_uniqueness_constraint_on_phone.up.sql, sql: alter table auth.mfa_factors drop constraint if exists mfa_factors_phone_key;\ndo $$\nbegin\n    if exists (\n         select 1\n         from pg_indexes\n         where indexname = 'unique_verified_phone_factor'\n         and schemaname = 'auth'\n    ) then\n        execute 'alter index auth.unique_verified_phone_factor rename to unique_phone_factor_per_user';\n    end if;\nend $$;\n: ERROR: relation \"unique_phone_factor_per_user\" already exists (SQLSTATE 42P07)","time":"2024-10-23T07:06:12Z"}
```

Again, we manually added the serial of this migration to the schema history, which temporarily restored the auth module’s health.

However, the potential for these types of issues occurring again is concerning, especially when considering Supabase for a production environment. While we would gladly switch to a paid plan and use Supabase for production instead of self-hosting or exploring alternatives, these ongoing issues in our test system are making us hesitant.

Could the Supabase team provide more insight into the timeline for addressing this problem? As suggested by @blytheaw and others, making the migrations idempotent seems like a feasible solution to avoid similar errors in the future.

We are excited to continue using Supabase, and we look forward to resolving these issues so we can confidently upgrade to a paid plan.

Thank you for your attention to this matter, and kind regards,
Thor

**Comment by brett-minca** [comment]
We've also been running into problems with this, any time I find one I have manually add something to our "supabase-auth-fix" script that manually inserts the version in to auth.schema_migrations when there's a migration that can't be idempotently applied.  It doesn't happen often, but it is very frustrating when it does.

**Comment by Tigatok** [comment]
I encountered this after I moved my repo to a monorepo. I am not sure if that had any effect on things. I had to run `insert into auth.schema_migrations values ('20221208132122');` and things are solved for me.

**Comment by Sriteja8Dev** [comment]
I am facing similar issue when verified the Authentication logs as suggested by [Kang Ming](https://github.com/kangmingtay)

{"level":"fatal","msg":"running db migrations: Migrator: problem creating schema migrations: could not execute CREATE TABLE \"schema_migrations\" (\n\"version\" VARCHAR (14) NOT NULL,\nPRIMARY KEY(\"version\")\n);\nCREATE UNIQUE INDEX \"schema_migrations_version_idx\" ON \"schema_migrations\" (version);: ERROR: relation \"schema_migrations\" already exists (SQLSTATE 42P07)","time":"2025-02-16T21:45:41Z"}

Firstly the table schema migrations is already present no idea why it is trying to create again.

when verified for the suggested fix , values '20221125140132', '20221208132122' are already present in the auth.schema_migrations table.

Kindly suggest some ways as this is blocking the further steps of my project.

**Comment by ghost** [comment]
The hack "insert into auth.schema_migrations values ('20221208132122');" is still relevant. Any idea how to handle it with the upcoming Supabase restrictions :

> On April 21, you will no longer be able to perform the following actions on the auth, storage, and realtime schemas:
> - Create tables and database functions
> - Drop existing tables or database functions
> - Create indexes on existing tables
> - Perform destructive actions (i.e. INSERT, UPDATE, DELETE, TRUNCATE) on the following migration tables:
>   - auth.schema_migrations
>   - storage.migrations
>   - realtime.schema_migrations

**Comment by aleix10kst** [comment]
We just had this issue, but we haven't changed anything. Do you have any idea when it will be fixed?

**Reply by stefan-girlich** [reply]
See [this comment](https://github.com/supabase/auth/issues/1570#issuecomment-2620167158) for the only solution that worked for me, which is deleting all Supabase Docker volumes:
```docker volume rm $(docker volume ls -q | grep supabase)```

Good to have a workaround, but it's not ideal that I have to apply it after every update of the Supabase CLI.

**Comment by ChronSyn** [comment]
Hitting this issue right now when trying to run locally via `npx supabase start`. I've already added `--exclude storage-api` (without it, the entire stack won't even start), but when checking the logs for Kong after trying to access an auth-based URL (e.g. refreshing an auth token), I get:

```
1114#0: *33 connect() failed (111: Connection refused) while connecting to upstream, client: <my IP>, server: kong, request: "GET /auth/v1/token HTTP/1.1", upstream: "http://<my VM ip>:9999/token", host: "<my VM ip>:54321"
```
If I try to send the request directly to the auth service in Insomnia (POST to `http://<my VM ip>:9999/token?grant_type=refresh_token`), I get "Error: Couldn't connect to server".

The docker logs from auth don't report any errors. It's literally just Kong which is saying it can't communicate with the upstream, with connection refused being the reason.

I can access the dashboard, and I can see my tables have been created correctly, but the issue is that I can't do anything auth-related such as creating a user (auto-confirmed or not).

The `insert into auth.schema_migrations values ('20221208132122');` suggestion is no longer possible to do from the Supabase dashboard due to changes made in April, so I have no idea if that's the fix.

I've tried this across several systems, and interestingly, it seems like the issue is primarily on Ubuntu systems I tried. Maybe it's a remote-development issue too, but if I run it locally on my Macbook Pro (i.e. not via a VM), then it works without issue.

However, what's really strange is that just a few weeks ago, the stack was working just fine on an Ubuntu 22.04 system, but trying to get back to working on it recently, and this issue showed up.

I've had some recent network changes (i.e. network router/firewall moved from pfSense to opnsense), so potentially that's a possible cause, but even when I try to run the CURL equivalent of an auth command on the system (via SSH terminal, but pointing the request to `127.0.0.1`), I still get 'connection refused' against the auth container.

It seems evident to me that there's either something problematic in my Docker setup on those systems that Supabase-cli specifically disagrees with, or there's a problem with Supabase-cli stack on Ubuntu.

**Comment by llorenspujol** [comment]
Yep folks, I confirm it’s over, we can’t use this solution anymore.

Running:

`insert into auth.schema_migrations values ('20221208132122');`

now results in the error:

**ERROR:  42501: permission denied for table schema_migrations**

**Comment by AtotheY** [comment]
any fixes? getting 503's only for auth, and the old temp fix doesn't seem to work? 

<img width="2650" height="1334" alt="CleanShot 2025-12-30 at 14 56 17@2x" src="https://github.com/user-attachments/assets/17e86801-a00e-415e-a057-323f3142ca19" />
